py2k/producer.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 ABSA Group Limited
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from confluent_kafka import SerializingProducer
+from confluent_kafka import KafkaError, Message
+
+
+class KafkaProducer:
+    def __init__(self, topic, producer_config):
+        self._topic = topic
+        self._producer = SerializingProducer(producer_config.dict)
+        self._serializer = producer_config.serializer
+
+    def produce(self, item):
+        while True:
+            try:
+                key, value = self._serializer.serialize(item)
+
+                self._producer.produce(
+                    topic=self._topic,
+                    key=key,
+                    value=value,
+                    on_delivery=self._delivery_report
+                )
+                self._producer.poll(0)
+                break
+            except BufferError as e:
+                print(
+                    f'Failed to send on attempt {item}. '
+                    f'Error received {str(e)}')
+                self._producer.poll(1)
+
+    def flush(self):
```

Comment:
Where do we call this?

Reply:
https://github.com/AbsaOSS/py2k/blob/feature/key-serialization-to-model/py2k/writer.py#L66